src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -1485,13 +1485,20 @@ static QVariant fcnRegexpReplace( const QVariantList &values, const QgsExpressio
   QString str = QgsExpressionUtils::getStringValue( values.at( 0 ), parent );
   QString regexp = QgsExpressionUtils::getStringValue( values.at( 1 ), parent );
   QString after = QgsExpressionUtils::getStringValue( values.at( 2 ), parent );
+  bool isValid_after{ values.at( 2 ).isValid() };
```

Comment:
Can you make that `isNul()`?
It will work with invalid and (valid) Null QVariants..